test/promote_literals_test.cpp

```diff
@@ -26,6 +26,7 @@
 #include <migraphx/program.hpp>
 #include <migraphx/dead_code_elimination.hpp>
 #include <migraphx/eliminate_common_subexpression.hpp>
+#include <migraphx/instruction.hpp>
```

Comment:
We dont use the instruction class here either, at least as far as I can tell. If windows has problems with incomplete types then maybe we should add this header in instruction_ref.hpp under `#ifdef _WIN32`